### PR TITLE
build: extract constant term from PauliSum

### PR DESCRIPTION
**Pull request by Boniface316**
## Description

Include description of feature this PR introduces or a bug that it fixes. Include the following information:

Simpler method to extract the constant value from PauliSum
## Please verify that you have completed the following steps

- [X] I have self-reviewed my code.
- [X] I have included test cases validating introduced feature/fix.

### CODE REVIEW

File: tests/orquestra/quantum/operators/_operators_test.py [hunk 1]
```diff
@@ -641,3 +641,14 @@ def test_paulisum_repr(self, pauli_sum):
     def test_paulisum_multiply_by_zero(self, pauli_sum: PauliSum):
         zero_op = pauli_sum * 0
         assert zero_op == PauliSum()
+
+    def test_paulisum_constant_value_zero(self, pauli_sum):
+        assert 0 == pauli_sum.constant_value
+
+    def test_paulisum_constant(self, pauli_sum):
```
Comment (mstechly) [inline tests/orquestra/quantum/operators/_operators_test.py:648]: Please do parameterize with other values as well – namely non-integers and negative values.

File: src/orquestra/quantum/operators/_pauli_operators.py [hunk 2]
```diff
@@ -549,6 +549,12 @@ def __hash__(self):
     def is_constant(self) -> bool:
         return len(self.terms) == 0 or all([term.is_constant for term in self.terms])
 
+    @property
+    def constant_value(self) -> Union[int, float]:
+        return sum(
```
Comment (mstechly) [inline src/orquestra/quantum/operators/_pauli_operators.py:554]: 1. It shouldn't be sum, there should be only one constant term 🤔  But fine, I see how software can be different from "in principle".
2. I think `Union[int, float]` doesn't really make sense, there is a different type for that, `Number` I think 🤔

File: src/orquestra/quantum/operators/_pauli_operators.py [hunk 3]
```diff
@@ -549,6 +549,12 @@ def __hash__(self):
...
+            [term.coefficient for term in self.terms if term.is_constant]  # type:ignore
```
Comment (mstechly) [inline src/orquestra/quantum/operators/_pauli_operators.py:555]: I don't like this `type:ignore` 🙃

File: tests/orquestra/quantum/operators/_operators_test.py [hunk 4]
```diff
@@ -641,3 +641,14 @@ def test_paulisum_repr(self, pauli_sum):
     def test_paulisum_multiply_by_zero(self, pauli_sum: PauliSum):
         zero_op = pauli_sum * 0
         assert zero_op == PauliSum()
+
+    def test_paulisum_constant_value_zero(self, pauli_sum):
+        assert 0 == pauli_sum.constant_value
```
Comment (mstechly) [inline tests/orquestra/quantum/operators/_operators_test.py:646]: [nitpick] Usually we do asserts the other way:
`assert pauli_sum.constant_value==0`

**Review by mstechly — CHANGES_REQUESTED** [review]

File: src/orquestra/quantum/operators/_pauli_operators.py (hunk 3 above)
Comment (Boniface316) [reply src/orquestra/quantum/operators/_pauli_operators.py:555]: I spent little over an hour dealing with this MyPy issue. @dexter2206?

File: tests/orquestra/quantum/operators/_operators_test.py (hunk 1 above)
Comment (Boniface316) [reply tests/orquestra/quantum/operators/_operators_test.py:648]: Done

File: tests/orquestra/quantum/operators/_operators_test.py (hunk 4 above)
Comment (Boniface316) [reply tests/orquestra/quantum/operators/_operators_test.py:646]: Done

**Review by mstechly — APPROVED** [review]

**Pull request merged**